sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RedirectStrategy.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+
+import java.util.Set;
+
+/**
+ * The interface for determining the {@link RedirectStrategy redirect strategy} used in {@link RedirectPolicy}.
+ */
+public interface RedirectStrategy {
+    /**
+     * Max number of redirect attempts to be made.
+     *
+     * @return The max number of redirect attempts.
+     */
+    int getMaxAttempts();
+
+    /**
+     * The header name to look up the value for the redirect url in response headers.
+     *
+     * @return the value of the header, or null if the header doesn't exist in the response.
+     */
+    String getLocationHeader();
+
+    /**
+     * The {@link HttpMethod http methods} that are allowed to be redirected.
+     *
+     * @return the set of redirect allowed methods.
+     */
+    Set<HttpMethod> getAllowedMethods();
+
+    /**
+     * Determines if the url should be redirected between each try.
+     *
+     * @param context the {@link HttpPipelineCallContext HTTP pipeline context}.
+     * @param httpResponse the {@link HttpRequest} containing the redirect url present in the response headers
+     * @param tryCount redirect attempts so far
+     * @param attemptedRedirectUrls attempted redirect locations used so far.
+     * @return {@code true} if the request should be redirected, {@code false} otherwise
+     */
+    boolean shouldAttemptRedirect(HttpPipelineCallContext context, HttpResponse httpResponse, int tryCount,
+                                  Set<String> attemptedRedirectUrls);
```

Comment:
Why do you pass in the context here - it doesn't seem to be used in the default strategy. Do you think some strategies might need it?

Reply:
Yes, we discussed this above here https://github.com/Azure/azure-sdk-for-java/pull/23617#discussion_r694176280.
We have got partner team feedback to have a way to access the context in other core policies. 
Ex - https://github.com/Azure/azure-sdk-for-java/issues/8504